src/irr/asset/CCPUMeshPacker.h

```diff
@@ -105,47 +104,58 @@ ReservedAllocationMeshBuffers CCPUMeshPacker<MDIStructType>::alloc(const Iterato
 	const size_t vtxCnt = std::accumulate(begin, end, 0ull, [](size_t init, ICPUMeshBuffer* mb) { return init + mb->calcVertexCount(); });
 
 	const uint32_t minIdxCntPerPatch = m_minTriangleCountPerMDIData * 3;
-	const uint32_t possibleMDIStructsNeededCnt = core::roundUp<uint32_t>(idxCnt, minIdxCntPerPatch);
+	
+	uint32_t possibleMDIStructsNeededCnt = 0u;
+	for (auto it = begin; it != end; it++)
+		possibleMDIStructsNeededCnt += ((*it)->getIndexCount() + minIdxCntPerPatch - 1) / minIdxCntPerPatch;
 
 	uint32_t MDIAllocAddr       = INVALID_ADDRESS;
 	uint32_t idxAllocAddr       = INVALID_ADDRESS;
 	uint32_t vtxAllocAddr       = INVALID_ADDRESS;
 	uint32_t perInsVtxAllocAddr = INVALID_ADDRESS;
 
-	//actually it will not work at all if m_MDIDataAlctrResSpc == nullptr or m_idxBuffAlctrResSpc == nullptr.. TODO
-	if (m_MDIDataAlctrResSpc)
+	MDIAllocAddr = m_MDIDataAlctr.alloc_addr(possibleMDIStructsNeededCnt, 1u);
+	if (MDIAllocAddr == INVALID_ADDRESS)
 	{
-		MDIAllocAddr = m_MDIDataAlctr.alloc_addr(possibleMDIStructsNeededCnt, 1u);
-		if (MDIAllocAddr == INVALID_ADDRESS)
-			return invalidReservedAllocationMeshBuffers;
+		_IRR_DEBUG_BREAK_IF(true);
+		return invalidReservedAllocationMeshBuffers;
 	}
+		
 	
-	if (m_idxBuffAlctrResSpc)
+	idxAllocAddr = m_idxBuffAlctr.alloc_addr(idxCnt, 1u);
+	if (idxAllocAddr == INVALID_ADDRESS)
 	{
-		idxAllocAddr = m_idxBuffAlctr.alloc_addr(idxCnt, 1u);
-		if (idxAllocAddr == INVALID_ADDRESS)
-			return invalidReservedAllocationMeshBuffers;
+		_IRR_DEBUG_BREAK_IF(true);
+		return invalidReservedAllocationMeshBuffers;
 	}
+		
 	
 	if (m_vtxBuffAlctrResSpc)
 	{
 		vtxAllocAddr = m_vtxBuffAlctr.alloc_addr(vtxCnt, 1u);
 		if (vtxAllocAddr == INVALID_ADDRESS)
+		{
+			_IRR_DEBUG_BREAK_IF(true);
 			return invalidReservedAllocationMeshBuffers;
+		}
+			
 	}
 	
 	if (m_perInsVtxBuffAlctrResSpc)
 	{
-		perInsVtxAllocAddr = m_perInsVtxBuffAlctr.alloc_addr(vtxCnt, 1u);
+		perInsVtxAllocAddr = m_perInsVtxBuffAlctr.alloc_addr(vtxCnt * m_vtxSize, 1u);
```

Comment:
why are you multiplying by the vertex size!?

Reply:
forgot to change that, it worked when `vtxSupportedSize` was in bytes